src/wallet/asyncrpcoperation_mergetoaddress.h

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef ASYNCRPCOPERATION_MERGETOADDRESS_H
```

Comment:
Attaching comment here about a reproducbile anchor problem when merging using "*" when there are both  utxos and notes present:

```
./zcash-cli z_mergetoaddress "*" ztJqA2UeXP5ZQWiuK8HRijfj3V7GxZNTXQkjf4WXc8Sj2Evgyen6DmeW5yhhZjTzFXhsqdS6bqXjV7oT9zu9GdzBqB6viRE 0.0001 0 0 "BEEF"
{
  "remainingUTXOs": 0,
  "remainingTransparentValue": 0.00000000,
  "remainingNotes": 0,
  "remainingShieldedValue": 0.00000000,
  "mergingUTXOs": 2,
  "mergingTransparentValue": 200.00000000,
  "mergingNotes": 1,
  "mergingShieldedValue": 1575.31036597,
  "opid": "opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea"
}
```
This fails:
```
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: z_mergetoaddress initialized (params={"fromaddresses":"*","toaddress":"ztJqA2UeXP5ZQWiuK8HRijfj3V7GxZNTXQkjf4WXc8Sj2Evgyen6DmeW5yhhZjTzFXhsqdS6bqXjV7oT9zu9GdzBqB6viRE","fee":0.00010000})
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: spending 1775.31036597 to send 1775.31026597 with fee 0.0001
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: transparent input: 200.00
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: private input: 1575.31036597
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: private output: 1775.31026597
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: fee: 0.0001
2018-01-25 21:17:48 opid-8e1ac86c-dc35-4465-b529-9a8fbd849aea: z_mergetoaddress finished (status=failed, error=Could not find previous JoinSplit anchor)
```
However, calling "ANY_ZADDR" at this point succeeds, as does "ANY_ZADDR".
Here is the "ANY_ZADDR" output:
```
./zcash-cli z_mergetoaddress "ANY_ZADDR" ztJqA2UeXP5ZQWiuK8HRijfj3V7GxZNTXQkjf4WXc8Sj2Evgyen6DmeW5yhhZjTzFXhsqdS6bqXjV7oT9zu9GdzBqB6viRE 0.0001 0 0 "BEEF"
{
  "remainingUTXOs": 0,
  "remainingTransparentValue": 0.00000000,
  "remainingNotes": 0,
  "remainingShieldedValue": 0.00000000,
  "mergingUTXOs": 0,
  "mergingTransparentValue": 0.00000000,
  "mergingNotes": 1,
  "mergingShieldedValue": 1575.31036597,
  "opid": "opid-deec2331-c1c2-4a75-9314-b72808b95255"
}
```
and log output:
```
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: z_mergetoaddress initialized (params={"fromaddresses":"ANY_ZADDR","toaddress":"ztJqA2UeXP5ZQWiuK8HRijfj3V7GxZNTXQkjf4WXc8Sj2Evgyen6DmeW5yhhZjTzFXhsqdS6bqXjV7oT9zu9GdzBqB6viRE","fee":0.00010000})
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: spending 1575.31036597 to send 1575.31026597 with fee 0.0001
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: transparent input: 0.00
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: private input: 1575.31036597
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: private output: 1575.31026597
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: fee: 0.0001
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: spending note (txid=1ae2ac96d5, vjoinsplit=0, ciphertext=0, amount=1575.31036597, height=652, confirmations=1)
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: generating note for change (amount=1575.31026597)
2018-01-25 21:22:10 opid-deec2331-c1c2-4a75-9314-b72808b95255: creating joinsplit at index 0 (vpub_old=0.00, vpub_new=0.0001, in[0]=1575.31036597, in[1]=0.00, out[0]=0.00, out[1]=1575.31026597)
2018-01-25 21:22:46 opid-deec2331-c1c2-4a75-9314-b72808b95255: Payment Disclosure: js=0, n=1, zaddr=ztkjJKs99AqHhoUUHa5h5cJ44tmQvBqgk66D2unWhk1xkP2L36SxhufEQLj5WyeS9uy8c5A4ZvpfEWuQkHE6PEdEuzUPkqt
2018-01-25 21:22:46 opid-deec2331-c1c2-4a75-9314-b72808b95255: Payment Disclosure: js=0, n=0, zaddr=ztJqA2UeXP5ZQWiuK8HRijfj3V7GxZNTXQkjf4WXc8Sj2Evgyen6DmeW5yhhZjTzFXhsqdS6bqXjV7oT9zu9GdzBqB6viRE
2018-01-25 21:22:46 Blockpolicy mempool tx 05b61e6592 not adding
2018-01-25 21:22:46 AddToWallet 05b61e6592fb92a4cd4497a2730009737fd2f9899974c66a195a2368c58edcae  new
2018-01-25 21:22:46 opid-deec2331-c1c2-4a75-9314-b72808b95255: z_mergetoaddress finished (status=success, txid=05b61e6592fb92a4cd4497a2730009737fd2f9899974c66a195a2368c58edcae)
```

Reply:
The above is also reproducible when manually specifying the array of addresses e.g. '["t....", "z....."]'